src/widget/widget.cpp

```diff
@@ -1540,12 +1554,20 @@ QString Widget::getStatusIconPath(Status status)
     }
 }
 
-inline QIcon Widget::getStatusIcon(Status status, uint32_t w/*=0*/, uint32_t h/*=0*/)
+inline QIcon Widget::getStatusIcon(Status status, uint32_t w, uint32_t h)
 {
-    if (w > 0 && h > 0)
-        return getStatusIconPixmap(status, w, h);
-    else
-        return QIcon(getStatusIconPath(status));
+#ifdef __linux__
+    QString desktop = getenv("XDG_CURRENT_DESKTOP");
+    if (desktop.isEmpty())
+        desktop = getenv("DESKTOP_SESSION");
+    desktop = desktop.toLower();
+    if (desktop == "xfce" || desktop.contains("gnome") || desktop == "mate") {
```

Comment:
not really, this if must cover window managers based on GTK2/3. `desktop!="unity"` will break support for i3, awesome and other wired managers.